extensions/amp-smartlinks/0.1/amp-smartlinks.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Services} from '../../../src/services';
+
+import {LinkRewriterManager} from './link-rewriter/link-rewriter-manager';
+import {Linkmate} from './linkmate';
+import {NRTV_CONFIG_ENDPOINT,
+  PAGE_IMPRESSION_ENDPOINT,
+  SMARTLINKS_REWRITER_ID} from './constants';
+import {getConfigOptions} from './linkmate-options';
+
+
+export class AmpSmartlinks extends AMP.BaseElement {
+  /**
+   * @param {!AmpElement} element
+   */
+  constructor(element) {
+    super(element);
+
+    /** @private {?../../../src/service/xhr-impl.Xhr} */
+    this.xhr_ = null;
```

Comment:
is the trailing underscore an AMP convention? Looks a bit weird. Would prefer `_privateFunc`

Reply:
it's amp convention